lib/options.js

```diff
@@ -12,90 +12,104 @@ module.exports = {
    * availableOptions - allowed values (only for type=enum)
    * description - human-readable description of the item
    * external - whether the option is settable via the API
+   * usage - array of supported types of usage (i.e. CONVERSION, VALIDATION)
    *
    * @param {string} [mode='document'] Describes use-case. 'document' will return an array
    * with all options being described. 'use' will return the default values of all options
+   * @param {Object} criteria Decribes required criteria for options to be returned. can have properties
+   *   external: <boolean>
+   *   usage: <array> (Array of supported usage type - CONVERSION, VALIDATION)
```

Comment:
usage should be a string and not an array IMO. As at a time, we would only want options related to either conversion or validation.

Reply:
Keeping it array is better solution, as it may be case in future where we need multiple usage type in criteria.